### DISCUSSION TITLE
I am wondering if the bepuphysics engine will work with source engine games

### DISCUSSION THREAD

**Discussion by Andru115** [bepu/bepuphysics2#332] [General]
Okay to begin, I just want to ask if it is possible to make the bepuphysics engine work with the source engine. I see that the physics engine valve made is very slow and laggy and this looks like an eager opportunity to replace it with a better engine, any help will be greatly appreciated!

**Comment by sayiliromer** [comment]
I have no experience on source engine but, you can compile bepuphysics as NativeAOT and use exported dll as a native dll from c++.
This is how i am using it in Unity. Although unity already is using C# it does not support CoreCLR it runs with Monoruntime. 

Here is how i did it;
https://github.com/bepu/bepuphysics2/discussions/331
